core/src/main/java/org/mskcc/cbio/portal/servlet/CancerStudyView.java

```diff
@@ -100,75 +123,195 @@ protected void processRequest(HttpServletRequest request, HttpServletResponse re
-            request.setAttribute(ERROR,
-                    "You are not authorized to view the cancer study with id: '" +
-                    cancerStudyIdentifier + "'. ");
-            return false;
-        }
-        else {
-            UserDetails ud = accessControl.getUserDetails();
-            if (ud != null) {
-                logger.info("CancerStudyView.validate: Query initiated by user: " + ud.getUsername());
-            }
-        }
-        
-        String sampleListId = (String)request.getAttribute(QueryBuilder.CASE_SET_ID);
-        if (sampleListId==null) {
-            sampleListId = cancerStudy.getCancerStudyStableId()+"_all";
-            request.setAttribute(QueryBuilder.CASE_SET_ID, sampleListId);
-        }
-        
-        SampleList sampleList = daoSampleList.getSampleListByStableId(sampleListId);
-        if (sampleList==null) {
-            request.setAttribute(ERROR,
-                    "Could not find sample list of '" + sampleListId + "'. ");
-            return false;
-        }
-        
-        request.setAttribute(QueryBuilder.CASE_IDS, sampleList.getSampleList());
-        
-        request.setAttribute(CANCER_STUDY, cancerStudy);
-        request.setAttribute(QueryBuilder.HTML_TITLE, cancerStudy.getName());
-        return true;
+                        Integer.parseInt(cancerStudyId));
+			}else{
+				if (accessControl.isAccessibleCancerStudy(cancerStudy.getCancerStudyStableId()).size() != 1) {
+					throw new NotAuthorizedException(cancerStudy.getCancerStudyStableId(), "unAuthorized");
+				}
+			}
+		} catch(NumberFormatException numberFormatException) {
+			LOG.warn("CancerStudyView.getCancerStudyDetails(): NumberFormatException = '" + numberFormatException.getMessage() + "'");
+		}
+    	return cancerStudy;
     }
     
-    private void setGeneticProfiles(HttpServletRequest request) throws DaoException {
-        CancerStudy cancerStudy = (CancerStudy)request.getAttribute(CANCER_STUDY);
-        GeneticProfile mutProfile = cancerStudy.getMutationProfile();
-        if (mutProfile!=null) {
-            request.setAttribute(MUTATION_PROFILE, mutProfile);
+    /**
+     * This method builds the response for the servlet request.
+     * Request format could be one of the following
+     * 1. Predefined cohort or a cohort list(Ex : cohorts=acc_tcga or cohorts=acc_tcga,brca_tcga)
+     * 2. User defined virtual cohort (Ex : cohorts=57e293a9d4c6c0c4ed706f2)
+     * 3. Any combination of 1 and 2
+     * 4. Encoded URI of study sample map string(Ex : {"acc_tcga":["TCGA-PK-A5H9-01","TCGA-OR-A5J3-01","TCGA-OR-A5JJ-01"]})
+     * @param request
+     * @return boolean value( whether success or not)
+     * @throws DaoException
+     * @throws JsonProcessingException
+     * @throws IOException
+     */
+    private boolean buildResponse(HttpServletRequest request) throws DaoException, JsonProcessingException, IOException {
```

Comment:
This method is a little long/complicated to analyze. Might be better to break down the functionality into several methods.